### PR TITLE
tests: for exit-code in case of error in config

### LINKED ISSUE #1980: webpack-cli@4 exits with 0 even when failing
# Bug report

`$ webpack` should exit with a non-0 code when it finds an error in `webpack.config.js`

**What is the current behavior?**

`$ webpack` exits with `0` even if it throws a fatal error:

**If the current behavior is a bug, please provide the steps to reproduce.**

```sh
npm install -g webpack webpack-cli@4
echo '.' > webpack.config.js
webpack
```

Notice that `webpack`’s version is irrelevant. `webpack-cli` is the problem

**What is the expected behavior?**

`webpack-cli@3` worked correctly, it exited with `1`. 

**Other relevant information:**
webpack version: 4 or 5 (webpack version irrelevant)
webpack-cli version: 4.1.0
Node.js version: v13.8.0
Operating System: macOS 10.15.4
Additional tools:

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**

**Summary**
added a test-case for the following issue.

Fixes #1980

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

**Comment by fregante** [comment]
This doesn't test syntax errors in the config, which is what #11789 is about

File: test/config/error/config-error.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+        expect(stderr).toContain('Invalid configuration object');
+        expect(stderr).toContain(`"development" | "production" | "none"`);
+        expect(exitCode).toBe(2);
+    });
+
+    it('should throw syntax error and exit with non-zero exit code', () => {
+        const { stderr, exitCode } = run(__dirname, ['-c', resolve(__dirname, 'syntax-error.js')]);
+
+        expect(stderr).toContain(`SyntaxError: Unexpected token '.'`);
```
Comment (snitin315) [inline test/config/error/config-error.test.js:17]:
```suggestion
        expect(stderr).toContain('SyntaxError: Unexpected token');
```

**Review by evilebottnawi — COMMENTED** [review]
We need to add test with broken syntax in config

**Review by evilebottnawi — APPROVED** [review]

**Review by anshumanv — APPROVED** [review]

**Comment by evilebottnawi** [comment]
Thanks!

**Pull request merged**